src/ZODB/scripts/repozo.py

```diff
@@ -755,6 +746,92 @@ def do_recover(options):
             raise
 
 
+def do_incremental_recover(options, repofiles):
+    datfile = os.path.splitext(repofiles[0])[0] + '.dat'
+    log('Recovering (incrementally) file to %s', options.output)
+    with open(options.output, 'r+b') as outfp:
+        outfp.seek(0, 2)
+        initial_length = outfp.tell()
+    with open(datfile) as fp:
```

Comment:
Doesn't reopening the index risk disobeying `find_files` logic ? Especially, the `--date` option.